chainio/clients/builder.go

```diff
@@ -29,28 +29,93 @@ type BuildAllConfig struct {
 	PromMetricsIpPortAddress   string
 }
 
-// Clients is a struct that holds all the clients that are needed to interact with the AVS and EL contracts.
-// TODO: this is confusing right now because clients are not instrumented clients, but
-// we return metrics and prometheus reg, so user has to build instrumented clients at the call
-// site if they need them. We should probably separate into two separate constructors, one
-// for non-instrumented clients that doesn't return metrics/reg, and another instrumented-constructor
-// that returns instrumented clients and the metrics/reg.
```

Comment:
please leave this, we still haven't addressed it afaik.

Reply:
done